tests/unit/utils/archives_test.py

```diff
@@ -495,3 +497,54 @@ def test_decompress_unknown_compression_suffix():
         _decompress(pathlib.Path('test.zip.zip'))
     msg, = excinfo.value.args
     assert msg == "Couldn't detect a compression from suffix .zip."
+
+
+@pytest.mark.parametrize(
+    ('recursive', 'remove_top_level', 'expected_files'),
+    [
+        pytest.param(
+            False, False,
+            (
+                'toplevel',
+                os.path.join('toplevel', 'recursive.zip'),
+            ),
+            id='recursive_false_remove_finished_false',
+        ),
+        pytest.param(
+            True, False,
+            (
+                'toplevel',
+                os.path.join('toplevel', 'recursive.zip'),
+                os.path.join('toplevel', 'recursive'),
+                os.path.join('toplevel', 'recursive', 'singlechild'),
+                os.path.join('toplevel', 'recursive', 'singlechild', 'test.file'),
+            ),
+            id='recursive_true_remove_finished_false',
+        ),
+    ],
+)
+def test_extract_archive_destination_path_not_None_no_remove_top_level_no_remove_finished_twice(
```

Comment:
I'm not sure of I understand what you are testing here. the test would already pass without this PR, right?

As I don't think that it's worth much effort to test this thoroughly, I'd probably just check the calls to `TarFile.extract()` for using the expected arguments via `unittest.mock` (for the second `extract_archive()` call).